js/views/userPage/Store.js

```diff
@@ -244,14 +201,64 @@ export default class extends BaseVw {
         this.$el.toggleClass('listView');
 
         if (this.storeListings) {
-          this.storeListings.viewType = type;
+          // this.storeListings.viewType = type;
+          this.renderListings(this.fullRenderedCollection);
         }
       }
     } else if (type === 'list') {
       this.$el.addClass('listView');
     }
   }
 
+  showDataChangedMessage() {
+    if (this.dataChangePopIn && !this.dataChangePopIn.isRemoved()) {
+      this.dataChangePopIn.$el.velocity('callout.shake', { duration: 500 });
+    } else {
+      const refreshLink =
+        `<a class="js-refresh">${app.polyglot.t('userPage.store.popinRefreshLink')}` +
+        '</a>';
+
+      this.dataChangePopIn = this.createChild(PopInMessage, {
+        messageText: app.polyglot.t('userPage.store.listingDataChangedPopin',
+          { refreshLink }),
+      });
+
+      this.listenTo(this.dataChangePopIn, 'clickRefresh', () => (this.collection.fetch()));
+
+      this.listenTo(this.dataChangePopIn, 'clickDismiss', () => {
+        this.dataChangePopIn.remove();
+        this.dataChangePopIn = null;
+      });
+
+      this.$popInMessages.append(this.dataChangePopIn.render().el);
+    }
+  }
+
+  showShippingChangedMessage() {
+    if (this.shippingChangePopIn ||
+      (this.shippingChangePopIn && this.shippingChangePopIn.isRemoved())) {
```

Comment:
If this.shippingChangePopIn is false, wouldn't (this.shippingChangePopIn && this.shippingChangePopIn.isRemoved()) also always be false, meaning the second condition is never evaluated?